XGROUP DELCONSUMER and ENTRIESREAD

michael-grunder opened the pull request:
Refactor XGROUP and implement the new DELCONSUMER (Redis 6.2.0) and ENTRIESREAD (Redis 7.0.0) options.  Additionally, add a proper phpdoc block to the stub file.

See #2068

yatsukhnenko commented on redis_commands.c:
```diff
@@ -5743,52 +5743,89 @@ int redis_xclaim_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
+    ZEND_PARSE_PARAMETERS_END_EX(return FAILURE);
+
+    /* Find the subcommand */
+    for (size_t i = 0; i < sizeof(xgroup_ops)/sizeof(xgroup_ops[0]); i++) {
```
IMO six `if/else` statements would be more obvious and simple and don't need ephemeral structs...

michael-grunder replied:
OK I removed my cool inline structure 😭 

(I like this better too)

michael-grunder committed:
XGROUP DELCONSUMER and ENTRIESREAD

Refactor XGROUP and implement the new DELCONSUMER (Redis 6.2.0) and
ENTRIESREAD (Redis 7.0.0) options.  Additionally, add a proper phpdoc
block to the stub file.

See #2068
